test/gateway/ost_prime/constructor.js

```diff
@@ -0,0 +1,81 @@
+const OSTPrime = artifacts.require("OSTPrime")
+  , BN = require('bn.js')
+  , web3 = require('../../test_lib/web3.js');
+
+const Utils = require('../../../test/test_lib/utils');
+
+const NullAddress = "0x0000000000000000000000000000000000000000";
+contract('OSTPrime.sol', function (accounts) {
+
+  describe('Construction', async () => {
+
+    const TOKEN_SYMBOL = "ST";
+    const TOKEN_NAME = "Simple Token";
+    const TOKEN_DECIMALS = new BN(18);
+
+    let brandedTokenAddress, ostPrime;
+
+    beforeEach(async function () {
+      brandedTokenAddress = accounts[2];
+    });
+
+    it('should pass with right set of parameters', async function () {
+
+      ostPrime = await OSTPrime.new(brandedTokenAddress);
+
+      assert.strictEqual(
+        web3.utils.isAddress(ostPrime.address),
+        true,
+        "OSTPrime contract address must not be zero"
```

Comment:
Is this required? Would `ostPrime = await OSTPrime.new(brandedTokenAddress);` not fail?

Reply:
Even if `ostPrime = await OSTPrime.new(brandedTokenAddress);` fails, tests will continue as there is no asserts.
This is helpful to assert that deployment is failed rather than getting the errors from the other test failure.

Reply:
You are correct, it fails 👍